File: corporatetaskgenerator.client/src/models/Task.ts
import { TaskPriority } from "./TaskPriority";
import { TaskStatus } from './TaskStatus';

export class Task {
    id: number;
    title: string;
    description?: string;
    priority: TaskPriority;
    dueDate: Date | null;
    status: TaskStatus;
    userId?: number;
    constructor(
        id: number,
        title: string,
        priority: TaskPriority,
        dueDate: Date,
        status: TaskStatus,
        description?: string,
        userId?: number
    ) {
        this.id = id;
        this.title = title;
        this.priority = priority;
        this.dueDate = dueDate;
        this.status = status;
        this.description = description;
        this.userId = userId;
    }
}

File: CorporateTaskGenerator.Server/Models/Task.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using CorporateTaskGenerator.Server.Enums;
using TaskStatus = CorporateTaskGenerator.Server.Enums.TaskStatus;

namespace CorporateTaskGenerator.Server.Models
{
    public class Task
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }

        [Required]
        [MaxLength(200)]
        public string Title { get; set; } = string.Empty;

        [Required]
        public TaskPriority Priority { get; set; }

        [Required]
        public DateTime DueDate { get; set; }

        [Required]
        public TaskStatus Status { get; set; }

        public string? Description { get; set; }

        // Foreign key to User
        [Required]
        public int UserId { get; set; }

        [ForeignKey(nameof(UserId))]
        public User? User { get; set; }
    }
}